CMakeLists.txt

```diff
@@ -33,6 +36,7 @@ if(NOT APPLE)
   option(OIDN_DEVICE_HIP "Enable HIP device." OFF)
 elseif(OIDN_ARCH STREQUAL "ARM64")
   option(OIDN_DEVICE_METAL "Enable Metal device." OFF)
+  option(OIDN_METAL_IOS "Enable iOS Metal device." OFF)
```

Comment:
We don't need an option for this. If we're building for iOS, the `IOS` CMake variable is set, which we can check instead.

Reply:
Gotcha, will remove the flag. I added it because I figured manually setting the `SYS_ROOT` and having cmake configure the target might be cumbersome, hence the sysroot_config.cmake file. I'll strip the symbol and have conditionals evaluate `DCMAKE_SYSTEM_NAME` for IOS

Reply:
It's sufficient to just evaluate the `IOS` CMake variable, existing code already does that. You don't need to manually set the sysroot if you set `CMAKE_SYSTEM_NAME=iOS`. That's the only CMake that needs to be set by the user to build for iOS.

Reply:
This should now be addressed. The `SYSROOT` variable is evaluated and `CMAKE_SYSTEM_NAME` is set accordingly, and later evaluated to properly configure gpu device building.